src/mongroup.cpp

```diff
@@ -109,7 +110,7 @@ MonsterGroupResult MonsterGroupManager::GetResultFromGroup(
     const mongroup_id &group_name, int *quantity, bool *mon_found )
 {
     const MonsterGroup &group = GetUpgradedMonsterGroup( group_name );
-    int spawn_chance = rng( 1, group.freq_total ); //Default 1000 unless specified
+    int spawn_chance = rng( 1, group.event_adjusted_freq_total() );
```

Comment:
This gets called quite a lot and has caused performance issues in the past in mapgen, we need a quick check that it's not dramatically increasing mapgen times.